resources/views/dashboard/training/statistics.blade.php

```diff
@@ -0,0 +1,100 @@
+            </div>
+            <div class="card text-center">
+                <div class="card-header">Unique Students</div>
+                <div class="card-body">
+                    <h2>{{ $stats['uniqueStudents'] }}</h2>
+                </div>
+            </div>
+            <div class="card text-center">
+                <div class="card-header">OTS Pass/Fail</div>
+                <div class="card-body">
+                    <h2><span class="text-success">{{ $stats['otsPerMonth']['pass'] }}</span> / <span class="text-danger">{{ $stats['otsPerMonth']['fail'] }}</span></h2>
+                </div>
+            </div>
+            <div class="card text-center">
+                <div class="card-header">Student<br />Cancellations</div>
+                <div class="card-body">
+                    <h2>{{ $stats['studentCancel'] }}</h2>
+                </div>
+            </div>
+            <div class="card text-center">
+                <div class="card-header">Student<br />No-Shows</div>
+                <div class="card-body">
+                    <h2>{{ $stats['studentNoShow'] }}</h2>
+                </div>
+            </div>
+        </div>
+        <div class="col-5">
+            <div class="card">
+                {{ Html::image('/dashboard/training/statistics/graph?id=1&year=' . $stats['dateSelect']['year'] . '&month=' . $stats['dateSelect']['month'], 'Sessions per month') }}
+            </div>
+            <br />
+            <div class="card">
+                {{ Html::image('/dashboard/training/statistics/graph?id=2&year=' . $stats['dateSelect']['year'] . '&month=' . $stats['dateSelect']['month'], 'Average session duration') }}
+            </div>
+        </div>
+        <div class="col-5">
+            <div class="card">
+                {{ Html::image('/dashboard/training/statistics/graph?id=3&year=' . $stats['dateSelect']['year'] . '&month=' . $stats['dateSelect']['month'], 'Sessions by instructor and type') }}
+            </div>
+            <br />
+            <div class="card">
+                {{ Html::image('/dashboard/training/statistics/graph?id=4&year=' . $stats['dateSelect']['year'] . '&month=' . $stats['dateSelect']['month'], 'Students requiring training') }}
+            </div>
+        </div>
+    </div>
+    <div class="row">
+        <div class="col-12">
+            <div class="card">
+                <div class="card-header">TA's Monthly Report
+                    <div class="float-right"><i class="fas fa-clipboard" onclick="copyToClipboard('taReport');"></i></div>
+                </div>
+                <div class="card-body">
+                    <textarea id="taReport" rows="3" class="form-control">{{ $stats['taMonthlyReport'] }}</textarea>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
+<script>
+    function copyToClipboard(eId) {
+        let e = document.getElementById(eId)
+        navigator.clipboard.writeText(e.innerHTML)
+    }
+</script>
```

Comment:
This function would be really nice as a helper in the JS files if we want to use it somewhere else in the future.

Reply:
Do you have a suggested location to add this? I'm still not very familiar with how laravel compiles JS

Reply:
I would add it to `resources/assets/js/helpers.js` and then you should be able to add `require ./helpers.js` in `app.js` and then it should be accessible after you run "npm run dev" to compile the assets (the pipeline will compile them for prod when it builds). Then it should be accessible in the view since we import `public/js/app.js` or something along the lines of that path (it exists somewhere in there) in the template.

Reply:
Done. Note that because of the cloudflare resource caching it may take a while to refresh the app.js resource, so this may not work for a few hours

Reply:
You should be able to go into cloudflare and dump the cache. But this is an admin tool so probably not the biggest deal if it doesn't work immediately.